pkg/util/interrupt.go

```diff
@@ -0,0 +1,95 @@
+/*
+Derived from:
+
+https://github.com/kubernetes/kubernetes/blob/master/pkg/util/interrupt/interrupt.go
+
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+)
+
+// childSignals are the allowed signals that can be sent to children in Unix variant OS's
+var childSignals = []os.Signal{syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT}
```

Comment:
again, deferring to @smarterclayton's responses on origin before making this not align with that.

for SIGQUIT.. the default go behavior on a SIGQUIT is to dump stack and exit.  that is the behavior i am attempting to preserve here:  remove the container (like any other exit), dump the stack, and exit.

if we didn't handle sigquit, we wouldn't get the removal behavior.  and if we trap the sigquit, then we have to explicitly dump the stack because go won't do it anymore.

not sure what you mean by "if we wouldn't wait for sigquit for perform clean up"